content/projects/gsoc/students.adoc

```diff
@@ -50,31 +81,181 @@ We highly recommend to make some contributions to the project while you work on
+. Setup your computer and your development environment to work on the project (see <<UsefulLinks>>).
+. Learn and discuss the process with the mentors:
+  * Setup the github project,
+  * Pull-requests,
+  * Code reviews.
+. Learn the coding style expectations from the mentors and/or from existing Jenkins code:
+  * proper spacing,
+  * proper indentation,
+  * proper capitalization,
+  * proper class, member and method names,
+  * proper annotations,
+  * unit and integration testing in Jenkins,
+  * the release process (for plugin projects).
+. Become familiar with the link:https://issues.jenkins-ci.org/secure/Dashboard.jspa[Jenkins issue tracker] (we use Jira)
+  * We will use Jira to track tasks in each coding period
+
+== Coding periods
+
+Students are expected to...
+
+. Work on the GSoC project as it is a full-time employment.
+ * It means that 30..40 hours per week is an **expected** workload though it can be adjusted upon the agreements with mentors.
+ * Push code to github almost every day of every coding period.
+ * Chat a line or two about what you are doing, almost every coding day, in your project channel (writing code, writing tests, updating documentation, etc.).
+ ** Just saying "Hi, today I am working on these classes" or "writing tests for ..." is good enough, but you can of course interact more as needed.
+ * Write a short summary of the work done each week, published to a personal blog or to the Jenkins website blogs.
+ ** A paragraph or two should be enough.
+ ** It's okay to say things like "_this_ and _that_ were challenging because _reason_".
+ * Interact with the community in a timely fashion when you need help (do not stay stuck without telling mentors).
+ * Say something when you are stuck, lost in the code, confused about the objectives, etc.
+ * Produce good quality code with reasonable amount of testing and documentation.
+ * Have a finalized deliverable at the end of the project.
+ ** For plugin development projects, this means releasing a plugin to the alpha or to the official update center.
+. Take Time off
+ * You have approximately 5 "vacation days" during the project, do not hesitate to use them if required.
+ * Notify your mentors in advance when you take time off.
+ * Use weekends to have a rest, avoid significant overwork and enjoy coding
+ * Timely notify mentors in the case of emergencies and outages (missing scheduled meetings, etc.).
+ * Timely notify mentors and org admins about unexpected time commitments (life goes on, it is normal - mentors will let you know if they can't be reached too).
+. Be present on-line
+ * Be around in _#jenkins_ IRC and in the project chats during the working hours (the link:https://gitter.im/jenkinsci/gsoc-sig[Jenkins GSoC Gitter Chat], and the Gitter Chat of your project)
+ * Be proactive; reach out to the community if required
+ * Optional: Attend Jenkins governance meetings if the timezone allows
```

Comment:
@martinda _Just a thought: Perhaps action items can be split into a checklist format?_

**Required _checklist_**
☑Be in Jenkins GSoC Gitter Chat
☐ Be in Gitter Chat of your project
☐ Be proactive; reach out to the community if required

**Optional _checklist_**
☐ Be around in _#jenkins_ IRC
☐ Attend Jenkins governance meetings if the timezone allows

**Not expected _checklist_**
☒ Strictly follow the originally submitted mini-design and the project proposal
☒ Investigate and solve *every* issue on your own

_( a student can print this document, and mark completed items )_